src/core/actions/unitactions/Convert.java

```diff
@@ -22,19 +23,31 @@ public Unit getTarget() {
 
     @Override
     public LinkedList<Action> computeActionVariants(final GameState gs) {
-        //TODO: compute all the Convert actions that are possible
-        return null;
+        LinkedList<Action> converts = new LinkedList<>();
+        Board b = gs.getBoard();
+        if(isFeasible(gs)){
+            Convert a = new Convert(this.unit, this.target);
```

Comment:
this.target wouldn't be set here. We need to return an action for every unit that super.unit can convert